src/indicators/money_flow_index.rs

```diff
@@ -167,61 +174,45 @@ mod tests {
     fn test_next_bar() {
         let mut mfi = MoneyFlowIndex::new(3).unwrap();
 
-        // tp = 2.0
         let bar1 = Bar::new().high(3).low(1).close(2).volume(500.0);
-        assert_eq!(mfi.next(&bar1), 50.0);
+        assert_eq!(round(mfi.next(&bar1)), 50.0);
 
-        // tp = 2.2, fm = 2.2*1000 = 2200, abs_total = 2200, pos_total = 2200
         let bar2 = Bar::new().high(2.3).low(2.0).close(2.3).volume(1000.0);
-        assert_eq!(mfi.next(&bar2), 100.0);
+        assert_eq!(round(mfi.next(&bar2)), 100.0);
 
-        // tp = 8.0, fm = 8*200 = 1600, abs_total = 3800, pos_total = 3800
         let bar3 = Bar::new().high(9).low(7).close(8).volume(200.0);
-        assert_eq!(mfi.next(&bar3), 100.0);
+        assert_eq!(round(mfi.next(&bar3)), 100.0);
 
-        // tp = 4.0, fm = -4.0*500 = -2000, abs_total = 5800 , pos_total = 3800
         let bar4 = Bar::new().high(5).low(3).close(4).volume(500.0);
-        assert_eq!(mfi.next(&bar4), 3800.0 / 5800.0 * 100.0);
+        assert_eq!(round(mfi.next(&bar4)), 65.517);
 
-        // tp = 3.0, fm = -3 * 5000 = -15000, abs_total = 5800+15000-2200=18600, pos_total=3800-2200=1600
         let bar5 = Bar::new().high(4).low(2).close(3).volume(5000.0);
-        assert_eq!(mfi.next(&bar5), 1600.0 / 18600.0 * 100.0);
+        assert_eq!(round(mfi.next(&bar5)), 8.602);
 
-        // tp = 1.5, fm = -1.5*6000= -9000, abs_total=18600+9000-1600=26000, pos_total=0
         let bar6 = Bar::new().high(2).low(1).close(1.5).volume(6000.0);
-        assert_eq!(mfi.next(&bar6), 0.0 / 23800.0 * 100.0);
+        assert_eq!(round(mfi.next(&bar6)), 0.0);
 
-        // tp = 2, fm = 2*7000=14000, abs_total=26000+14000-2000=38000, pos_total=14000
         let bar7 = Bar::new().high(2).low(2).close(2).volume(7000.0);
-        assert_eq!(mfi.next(&bar7), 14000.0 / 38000.0 * 100.0);
```

Comment:
Could you please keep the old comments and the expect result (like `14000.0 / 38000.0 * 100.0`).
Without it it's practically impossible to reason about the test expectations

Reply:
Since we changed our logic, I will have to update the comments.

But the real problem is that the previous tests was hiding a floating point error.

```rust
        // tp = 8.0
        // raw_money_flow = 8.0*200 = 1600
        // total_positive_money_flow = 2199.9999999999995+1600 = 3799.9999999999995
        // total_negative_money_flow = 0+0
        // total_money_flow = 3799.9999999999995+0 = 3799.9999999999995
        let bar3 = Bar::new().high(9).low(7).close(8).volume(200.0);
        let result = mfi.next(&bar3);
        assert_eq!(result, 3799.9999999999995 / 3800.0 * 100.0); // <= fails
        assert_eq!(result, 3799.9999999999995 / 3799.9999999999995 * 100.0); // <= works

        // tp = 4.0
        // raw_money_flow = 4.0*500 = 2000
        // total_positive_money_flow = 2199.9999999999995+1600+0 = 3799.9999999999995
        // total_negative_money_flow = 0+0+2000 = 2000
        // total_money_flow = 3799.9999999999995+2000 = 5799.9999999999995
        let bar4 = Bar::new().high(5).low(3).close(4).volume(500.0);
        let result = mfi.next(&bar4);
        assert_eq!(result, 3799.9999999999995 / 5800.0 * 100.0); // <= works
        assert_eq!(result, 3799.9999999999995 / 5799.9999999999995 * 100.0); // <= works
```

Since we removed the usage of `total_absolute_money_flow`, we have a better floating point precision.
The `assert_eq` trick still works for the test with `bar4` and others but not anymore for `bar3`.

Is it ok if I change the test with only exacts values ?
I tested the previous and the new MFI versions and they are exactly the same (except some some tricky floating values like `bar3`), so changing the values should not hide any regression.

But, do we really want to comment the implementation of the function inside a test ?